src/main/kotlin/com/hibob/academy/employee_feedback_feature/resource/FeedbackResource.kt

```diff
@@ -53,7 +58,60 @@ class FeedbackResource(private val feedbackService: FeedbackService) {
         return if (hasPermission(employeeRole, Permissions.VIEW_ALL_FEEDBACKS)) {
             Response.ok(feedbackService.getAllFeedbacks(companyId)).build()
         } else {
-            Response.status(Response.Status.FORBIDDEN).entity("UNAUTHORIZED: You do not have access to view feedbacks").build()
+            Response.status(Response.Status.UNAUTHORIZED)
+                .entity("UNAUTHORIZED: You do not have access to view feedbacks")
+                .build()
         }
     }
+
+    @Path("view/filter")
+    @POST
+    @Produces(MediaType.APPLICATION_JSON)
+    fun getFilteredFeedbacks(@Context requestContext: ContainerRequestContext, filter: FeedbackFilter): Response {
+        val companyId = extractClaimAsLong(requestContext, "companyId")!!
+        val employeeRole = extractRole(requestContext)!!
+
+        return if (hasPermission(employeeRole, Permissions.VIEW_ALL_FEEDBACKS)) {
```

Comment:
return with if - it's not clear.

also - first do the permission validation and return not authorised.
Then you can continue to the logic itself, without an if.

Or, another option, hasPermissions can throw the unauthorised exception, and then you don't need this if